### PR TITLE
Support torch.load with encoding

### PR DESCRIPTION
**Pull request by ssnl**
Addresses a common compatibility issue when loading Py2 checkpoints in Py3 regarding to bytes.

E.g., 
[1] https://github.com/pytorch/pytorch/issues/5994,
[2] https://github.com/CSAILVision/places365/issues/25,
[3] https://discuss.pytorch.org/t/how-to-load-a-saved-model-trained-on-pytorch-0-3-1-python-2-7-on-pyorch-1-0-python-3-7/31212

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -7915,6 +7917,34 @@ def test_serialization(self):
                 f.seek(0)
                 c = torch.load(handle)
             self._test_serialization_assert(b, c)
+        # test non-ascii encoding of bytes arrays/strings
+        # The following bytes are produced by serializing
+        #   [u'żąąóżąż'.encode('utf-8'), torch.zeros(1, dtype=torch.float), 2]
```
Comment (soumith) [inline test/test_torch.py]: fbcode restrictions dont allow non-ascii characters in files, i think. i'll have to import and modify, can you just modify it here instead

Comment (SsnL) [reply test/test_torch.py]: Sure. I'll fix it here.

Comment (SsnL) [reply test/test_torch.py]: Fixed :)

Comment (apaszke) [reply test/test_torch.py]: Interesting, it produces pretty much exactly the set of special characters for Polish 🤔

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**